src/main/java/com/exadel/ipromise/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,81 @@
+package com.exadel.ipromise.service.impl;
+
+import com.exadel.ipromise.converter.UserConverter;
+import com.exadel.ipromise.dao.UserDao;
+import com.exadel.ipromise.dto.UserDto;
+import com.exadel.ipromise.entity.User;
+import com.exadel.ipromise.service.UserService;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.server.ResponseStatusException;
+
+import javax.servlet.http.HttpSession;
+
+@Service("UserService")
+public class UserServiceImpl implements UserService {
+
+    String USER_ALREADY_EXISTS = "User with this e-mail already exists";
+    String USER_NOT_EXIST = "User with this e-mail does not exist";
+    String WRONG_LOGIN_OR_PASSWORD = "Wrong login or password!";
+
+    private final UserConverter userConverter;
+    private final UserDao userDao;
+
+    public UserServiceImpl(UserConverter userConverter, UserDao userDao) {
+        this.userConverter = userConverter;
+        this.userDao = userDao;
+    }
+
+    @Override
+    @Transactional
+    public UserDto addUser(UserDto userDto, HttpSession session) {
+
+        User user = userConverter.convertToEntity(userDto);
```

Comment:
Don't forget about validation. We cannot trust all data from client

Reply:
This functionality will be implemented in the next PR